students/Josh_HOff/lesson04/mailroom2.py

```diff
@@ -0,0 +1,79 @@
+import pathlib
+import os
+donors_list = {'Ralph Anders': [5, 10], 'Andrei Wasinski': [101, 151, 75], 'Stalk Holmes': [40], 'Traci Johnston': [20], 'James Hendrick': [60], 'Angelica Kisel': [45, 25, 55.60]}
+donor = ''
+donation = 0
+tab = '    '
+
+#this function gets a donor's name and donation amount and adds them both to the donor list.
+def send_thank_you():
+    while True:
+        print('')
+        donor = input('Type a Full Name: ')
+        if switch_func_dict.get(donor, continue_func)() == False:
+            return
+        elif donor != 'list':
+            print('')
+            donation = input('What is the donation amount?: ')
+            if donation == 'quit':
+                return
+            donation = float(donation)
+            if donor in donors_list:
+                donors_list[donor] += [donation]
+            else:
+                donors_list[donor] = [donation]
+            print('')
+            print(f'Hello, {donor}! Thank you very much for your generous donation of ${donation:.2f}! Your contribution is essential and will be well utilized.')
+            
+#this function prints out a list of the donors in the donor list.            
+def print_list():
+    print('')
+    for i in donors_list:
+        print(i)
+    return True
+
+#this function prints out a report on the prompt screen.    
+def create_report():
+    print('')
+    y = '|'
+    print(f'Donor Name{y:>14} Total Given {y} Num Gifts {y} Average Gift')
+    print('-' * 63)
+    temp_set = donors_list.copy()
+    for i, val in donors_list.items():
+        temp_set[i] = sum(val)        
+    temp_set = dict(sorted(temp_set.items(), key=lambda kv: kv[1], reverse=True))
+    for i, val in temp_set.items():
+        gift = len(donors_list[i])
+        average = ( val / gift)
+        print(f'{i:<23} $ {val:>11.2f} {gift:>11} {average:>11.2f}')
```

Comment:
👍 on using sorted

we can simplify the code further, replace lines 41 through 45 with:
```
    # no need for temp list to compute the sum, you can do it on the fly here
    sorted_donors = sorted(donors_list.items(), key=lambda k: sum(k[1]), reverse=True)
    # now loop directly over the sorted structure
    for donor_name, donations in sorted_donors:
        pass
```

Reply:
Got it, I've implemented it in my code.